src/call.c

```diff
@@ -2102,7 +2103,20 @@ static void sipsess_close_handler(int err, const struct sip_msg *msg,
 		xfer_cleanup(call, reason);
 
 	call_stream_stop(call);
-	call_event_handler(call, CALL_EVENT_CLOSED, "%s", reason);
+
+	if (msg) {
+		reason_hdr = sip_msg_hdr(msg, SIP_HDR_REASON);
+	}
+
+	if (reason_hdr) {
+		info("Cancel reason: %r\n", &reason_hdr->val);
+		call_event_handler(call, CALL_EVENT_CLOSED, "%s | %r",
+						   reason, &reason_hdr->val);
```

Comment:
Maybe we should use `"%s,%r"`.

Reply:
I wanted that string to be parseable again in the upper layers. One could still look for `SIP ;cause=` and start parsing there if a comma is used as a separator. It just felt better to have the `|` in there so that upper layers know that there is a reason to be parsed in that string.

If you prefer the comma I'll change it, just let me know :)

Reply:
I'll change it to use the comma. Using json seems complicated, since I'd need to escape any quotes in the enclosed strings. Thanks for the quick feedback :)